backend/src/apiserver/webhook/pipelineversion_webhook.go

```diff
@@ -119,14 +119,77 @@ func (p *PipelineVersionsWebhook) ValidateDelete(_ context.Context, _ runtime.Ob
 	return nil, nil
 }
 
-func NewPipelineVersionWebhook(client ctrlclient.Client) (http.Handler, error) {
+func (p *PipelineVersionsWebhook) Default(ctx context.Context, obj runtime.Object) error {
+	pipelineVersion, ok := obj.(*k8sapi.PipelineVersion)
+	if !ok {
+		return &apierrors.StatusError{
+			ErrStatus: metav1.Status{
+				Code:    http.StatusBadRequest,
+				Reason:  metav1.StatusReasonBadRequest,
+				Message: fmt.Sprintf("expected a PipelineVersion object but got %T", pipelineVersion),
+			},
+		}
+	}
+
+	pipeline := &k8sapi.Pipeline{}
+	nsName := types.NamespacedName{Namespace: pipelineVersion.Namespace, Name: pipelineVersion.Spec.PipelineName}
+
+	err := p.Client.Get(ctx, nsName, pipeline)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return newBadRequestError("The spec.pipelineName doesn't map to an existing Pipeline object")
+		}
+
+		return err
+	}
+
+	if pipelineVersion.Labels == nil {
+		pipelineVersion.Labels = map[string]string{}
+	}
+
+	// Labels for efficient querying
+	pipelineVersion.Labels["pipelines.kubeflow.org/pipeline-id"] = string(pipeline.UID)
+	pipelineVersion.Labels["pipelines.kubeflow.org/pipeline"] = string(pipeline.Name)
+
+	for i := range pipelineVersion.OwnerReferences {
+		ownerRef := &pipelineVersion.OwnerReferences[i]
+		if ownerRef.APIVersion != k8sapi.GroupVersion.String() || ownerRef.Kind != "Pipeline" {
+			continue
+		}
+
+		ownerRef.Name = pipeline.Name
+		ownerRef.UID = pipeline.UID
+
+		return nil
+	}
+
+	pipelineVersion.OwnerReferences = append(pipelineVersion.OwnerReferences, metav1.OwnerReference{
+		APIVersion: k8sapi.GroupVersion.String(),
+		Kind:       "Pipeline",
+		Name:       pipeline.Name,
+		UID:        pipeline.UID,
+	})
```

Comment:
an observation here, I'm wondering how this affects pipeline deletions, currently the db store implementation you can't delete the Pipeline without first having to delete the PipelineVersions, the use of ownerreferences here makes it a non issue for the k8s native store implementation, something to note for @rimolive maybe when doing pipeline deletions

Reply:
@HumairAK and @rimolive do you think we should set `blockOwnerDeletion` to prevent the pipeline from being deleted if the pipeline versions are still around? I lean towards no because I like the idea of being able to fully delete a pipeline with one API call but wanted your take.

Reply:
sigh, I agree with you but I think there has been discussion in the past about adding a separate api or toggle in the pipeline deletion call something like a parameter "cascade=true" to delete all at once, so in this case I'm leaning towards consistency and blocking it, but open to thoughts

Reply:
Okay, I added `blockOwnerDeletion`!